### ISSUE TITLE
Clangd support

### ISSUE THREAD

**Issue by b3n-h4il** [build2/build2#403]
Is it possible for build2 to generate a compile_commands.json file? I've seen tools like [compiledb](https://github.com/nickdiego/compiledb) and [bear](https://github.com/rizsotto/Bear), which generate a compile_commands.json file when the "build" command is issed, but they seem to work only with make based builds. Even with just a simple "clang++ hello.cpp", bear is capable of generating the file. But when using bear with the "b" command in a build2 project, the compile_commands.json file is empty. Is there a way to use clangd with build2?

**Comment by wroyca** [comment]
You can use [compiledb](https://github.com/helmesjo/compiledb/tree/support-cl-clang-cl) with `b -vn clean update |& compiledb`. The tool is unmaintained but work well enough from experience. For modules, see this patch: https://github.com/wroyca/compiledb/commit/de0aba35edf3f36fac6cbb8b005a233a2d45c0d5.patch

```
pip install git+https://github.com/helmesjo/compiledb@support-cl-clang-cl
```
or
```
pip install git+https://github.com/wroyca/compiledb@support-cl-clang-cl
```

Also related:
https://github.com/build2/build2/issues/96

**Comment by Norman-Normandy** [comment]
Have been waiting on this feature to be supported for a couple versions now.  There's also this online [tool](https://texttoolkit.com/compilation-database-generator) if your system doesn't have pip. Simply copy paste the compiler output and it generates a compile_commands.json file.

Would be nice to have a general time estimate on the progress.

**Comment by wroyca** [comment]
> Thanks @Norman-Normandy, I'm not the only one then. Bear is activelly maintained. I'd like to use it while B2 doesn't have the hability to generate the compile database. I've opened an [issue](https://github.com/rizsotto/Bear/issues/584) on the bear repo asking about B2 support and had some reply from the team. But I'm not sure about everything they're asking. I'd like your help in the discussion, if you're interested. Didn't have much time these past few days.

FYI, if Bear is not an option, compiledb may not be maintained anymore, but it works flawlessly (thus far) with every scenario I've thrown at it :)

**Comment by boris-kolpackov** [comment]
The compilation database support is now available in the latest [staged](https://build2.org/community.xhtml#stage) `build2` toolchain. There is also the [Compilation Database](https://build2.org/stage/build2/doc/build2-build-system-manual.xhtml#cc-compiledb) section in the manual with the documentation.

Compilation database is used in different scenarios (IDEs, static analysis, language servers, etc) so it would be helpful if everyone interested in this feature tried it and checked that it works for their use case.

**Comment by b3n-h4il** [comment]
Sorry for taking so long to answer here. Couldn't work in my project for a while. Tried this feature and it works; just used `config.cc.compiledb=.../<dir>/` where dir is my project's root folder, not the configuration folder, but the place where the manifests (like repositories.manifest) are. 

I'm using the [Pulsar-edit](https://pulsar-edit.dev) editor with the [pulsar-ide-clangd](https://web.pulsar-edit.dev/packages/pulsar-ide-clangd) extension. This extension uses clangd to get information about the project, but clangd won't look for the `compile_commands.json` file inside build directories, instead, it starts at the source directory and then looks in the parent folders until it finds the database build file. I haven't tried a lot of different scenarios, but my usecase works fine using only Build2 features.

**Comment by boris-kolpackov** [comment]
Looks like this works well in the staged toolchain, so I am going to close this issue.
